TCC.Lib/Helpers/FileExtensions.cs

```diff
@@ -7,6 +7,41 @@ namespace TCC.Lib.Helpers
 {
     public static class FileExtensions
     {
+        public static async Task Lock(this FileInfo lockFilePath, Func<Task> action)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this method into two, one handling parameters check and the other handling the asynchronous code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS4457)